dialects/transfer.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+from abc import ABC
+from z3 import *
+
+from dataclasses import dataclass, field
+from enum import Enum
+from xdsl.dialects.builtin import AnyIntegerAttr
+from xdsl.dialects.builtin import ArrayAttr
+from xdsl.dialects.builtin import IntegerAttr
+from xdsl.dialects.builtin import IndexType
+from xdsl.dialects.builtin import IntegerType
+from xdsl.dialects.builtin import ContainerOf
+from xdsl.dialects.builtin import TypeAttribute
+from xdsl.dialects.builtin import i1
+from xdsl.dialects.arith import Constant
+import functools
+from dataclasses import dataclass
+from enum import Enum
+from typing import Annotated, List, Sequence, TypeVar, Union, Set, Optional
+
+from xdsl.ir import (
+    ParametrizedAttribute,
+    Dialect,
+)
+
+from xdsl.irdl import (
+    AllOf,
+    OpAttr,
+    VarOpResult,
+    VarOperand,
+    VarRegion,
+    irdl_attr_definition,
+    attr_constr_coercion,
+    irdl_data_definition,
+    irdl_to_attr_constraint,
+    irdl_op_definition,
+    ParameterDef,
+    Operand,
+    OpResult,
+    IRDLOperation,
+    AnyOf,
+)
+from xdsl.utils.exceptions import VerifyException
+
+
+@irdl_attr_definition
+class AbstractValueType(ParametrizedAttribute, TypeAttribute):
+    name: str = "absValue"
```

Comment:
```suggestion
    name: str = "abs_value"
```
To remain consistent with the rest of the MLIR notations.